skill-tree/src/test/java/com/RDS/skilltree/unit/EndorsementServiceTest.java

```diff
@@ -602,4 +607,118 @@ void testCreateEndorsementWithInvalidSkill() {
         // Verify that save method is not called
         verify(endorsementRepository, never()).save(any(EndorsementModel.class));
     }
+
+    @Test
+    @DisplayName("Return unauthorized given user is not a super user")
+    public void itShouldReturnUnauthorizedIfUserIsNotSuperUser() {
+        UserModel userModel = new UserModel();
+        userModel.setRole(UserRole.USER);
+        when(auth.getPrincipal()).thenReturn(userModel);
+        SecurityContextHolder.getContext().setAuthentication(auth);
+
+        UUID endorsementId = UUID.randomUUID();
+        String status = EndorsementStatus.APPROVED.name();
+
+        AccessDeniedException exception =
+                assertThrows(
+                        AccessDeniedException.class,
+                        () -> endorsementService.updateEndorsementStatus(endorsementId.toString(), status));
+        assertEquals("Unauthorized access", exception.getMessage());
```

Comment:
There is a better way to assert exceptions - you can refer this https://junit.org/junit5/docs/current/user-guide/#writing-tests-assertions